test/e2e/test_helm_install_upgrade.go

```diff
@@ -120,10 +120,10 @@ func TestHelmUpgrade(t *testing.T) {
 					},
 				},
 				{
-					Name: "DataPlane deployment is patched after operator upgrade (due to change in default Kong image version to kong:3.8)",
+					Name: fmt.Sprintf("DataPlane deployment is patched after operator upgrade (due to change in default Kong image version to %q)", helpers.GetDefaultDataPlaneImagePreviousMinor()),
 					Func: func(c *assert.CollectT, cl *testutils.K8sClients) {
 						gatewayDataPlaneDeploymentIsPatched("gw-upgrade-onebeforelatestminor-latestminor=true")(ctx, c, cl.MgrClient)
-						gatewayDataPlaneDeploymentHasImageSetTo("gw-upgrade-onebeforelatestminor-latestminor=true", "kong:3.8")(ctx, c, cl.MgrClient)
+						gatewayDataPlaneDeploymentHasImageSetTo("gw-upgrade-onebeforelatestminor-latestminor=true", helpers.GetDefaultDataPlaneImagePreviousMinor())(ctx, c, cl.MgrClient)
```

Comment:
I do see the need to use `GetDefaultDataPlaneImage()` to make the test work in KGO EE test harness 👍 but I'm not completely on board with using `GetDefaultDataPlaneImagePreviousMinor()` and similar here.

The reason I'm raising this is that for instance, imaging next Kong version is released and [`DefaultDataPlaneTag`](https://github.com/Kong/gateway-operator/blob/dd397cf447c2d06b3ec44dbc5de116d5192ea543/pkg/consts/dataplane.go#L87-L94) is bumped.

At that point this test starts failing because 1.4.0's dp version didn't change but the test's expectation did.

My proposal here is to use `GetDefaultDataPlaneImage()` ( to make EE tests pass) but with a hardcoded tag in the expectation. 

WDYT?

Reply:
I see your point. It makes sense to me to hardcode the GW tag for a fixed KGO version which is the case in these tests. How about this: https://github.com/Kong/gateway-operator/pull/981/commits/ed2637d31b9b407a2074b210bf012d4d38178011?